components/remote_settings/src/context.rs

```diff
@@ -0,0 +1,123 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use serde_json::{Map, Value};
+
+/// Remote settings context object
+///
+/// This is used to filter the records returned. We always fetch all `records` from the
+/// remote-settings storage. Some records could have a `filter_expression`.  If this is passed in
+/// and the record has a `filter_expression`, then only returns where the expression is true will
+/// be returned.
+///
+/// See https://remote-settings.readthedocs.io/en/latest/target-filters.html for details.
+#[derive(Debug, Clone, Default, uniffi::Record)]
+pub struct RemoteSettingsContext {
+    /// The delivery channel of the application (e.g "nightly")
+    #[uniffi(default = None)]
+    pub channel: Option<String>,
+    /// User visible version string (e.g. "1.0.3")
+    #[uniffi(default = None)]
+    pub app_version: Option<String>,
+    /// String containing the XUL application app_id
+    #[uniffi(default = None)]
+    pub app_id: Option<String>,
+    /// The locale of the application during initialization (e.g. "es-ES")
+    #[uniffi(default = None)]
+    pub locale: Option<String>,
+    /// The name of the operating system (e.g. "Android", "iOS", "Darwin", "WINNT")
+    #[uniffi(default = None)]
+    pub os: Option<String>,
+    /// The user-visible version of the operating system (e.g. "1.2.3")
+    #[uniffi(default = None)]
+    pub os_version: Option<String>,
+    /// Form-factor of the device ("phone", "tablet", or "desktop")
+    #[uniffi(default = None)]
+    pub form_factor: Option<String>,
+    /// Country of the user.
+    ///
+    /// This can be populated in two ways:
+    ///   - The second component of the locale
+    ///   - By using the Mozilla geolocation service, which determines country via the user's IP.
```

Comment:
Nit: These doc comments are useful for consumers (like me), and for that reason this particular comment shouldn't be so prescriptive IMO. Desktop uses the `Region` module to get the client's country, and a desktop engineer might not know how that works under the hood. If they see that `Region` isn't included in this list, they might be confused. So I would make it clear that the two ways listed here are only examples, and this isn't an exhaustive list of how `country` can be populated.

(Also, IIRC desktop only uses the Mozilla location service on Nightly, and it uses a Google API in Release.)